src/settings/project/mod.rs

```diff
@@ -143,144 +139,116 @@ impl Manifest {
                             }
-                            wdd
-                        }
-                        None => {
-                            message::warn(deprecate_warning);
-                            true // workers_dot_dev defaults to true when it's top level and --release is not passed
                         }
                     }
+                    wdd
+                } else {
+                    message::warn(deprecate_warning);
+                    true
                 }
             }
 
             // environment configuration
-            Some(environment) => match environment.workers_dot_dev {
-                Some(wdd) => {
+            Some(environment) => {
+                if let Some(wdd) = environment.workers_dot_dev {
                     if wdd && environment.route.is_some() {
                         failure::bail!(wdd_failure)
                     }
                     wdd
-                }
-                None => {
-                    match self.workers_dot_dev {
-                        Some(wdd) => {
-                            if wdd && environment.route.is_some() {
-                                false // use route if workers_dot_dev = true is inherited
-                            } else {
-                                wdd // inherit from top level
-                            }
-                        }
-                        None => false,
+                } else if let Some(wdd) = self.workers_dot_dev {
+                    if wdd && environment.route.is_some() {
+                        false // allow route to override workers_dot_dev = true if wdd is inherited
+                    } else {
+                        wdd // inherit from top level
                     }
+                } else {
+                    false // if absent -> false
                 }
-            },
-        };
-
-        let kv_namespaces = match environment {
-            Some(environment) => match &environment.kv_namespaces {
-                Some(kv) => Some(kv.clone()),
-                None => None,
-            },
-            None => self.kv_namespaces.clone(),
-        };
-
-        let account_id = match environment {
-            Some(environment) => match &environment.account_id {
-                Some(a) => a.clone(),
-                None => self.account_id.clone(),
-            },
-            None => self.account_id.clone(),
+            }
         };
 
-        let name = match environment {
-            Some(environment) => match &environment.name {
-                Some(name) => {
-                    let name = name.clone();
-                    if name == self.name {
-                        failure::bail!(format!(
-                            "{} Each `name` in your wrangler.toml must be unique",
-                            emoji::WARN
-                        ))
+        let route = if let Some(environment) = environment {
```

Comment:
Same advice above can apply here--I think it's possible we can reduce nested matches :D